contracts/src/Gateway.sol

```diff
@@ -394,6 +416,49 @@ contract Gateway is IGateway, IInitializable {
         emit PricingParametersChanged();
     }
 
+    // @dev Register a new fungible Polkadot token for an agent
+    function _registerForeignToken(bytes32 agentID, address agent, bytes memory params) internal {
+        (bytes32 tokenID, string memory name, string memory symbol, uint8 decimals) =
+            abi.decode(params, (bytes32, string, string, uint8));
+        AssetsStorage.Layout storage $ = AssetsStorage.layout();
+        if ($.tokenRegistryByID[tokenID].isRegistered == true) {
+            revert TokenAlreadyRegistered();
+        }
+        ERC20 foreignToken = new ERC20(name, symbol, decimals);
+        address token = address(foreignToken);
+        TokenInfo memory info =
+            TokenInfo({isRegistered: true, isForeign: true, tokenID: tokenID, agentID: agentID, token: token});
+        $.tokenRegistry[token] = info;
+        $.tokenRegistryByID[tokenID] = info;
+        foreignToken.transferOwnership(agent);
```

Comment:
Instead of transferring ownership, just pass the agent address to the constructor of the ERC20:

```solidity
public address immutable owner; 

constructor(address _owner, ...) {
  owner = _owner
}
```

This saves gas, and reduces our reliance on OpenZeppelin libraries, which I really don't want to use for various reasons, including:
* I don't like OpenZepellins widespread use of inheritance, I think it makes auditing harder.

Reply:
https://github.com/Snowfork/snowbridge/pull/1155/commits/b00538f4ed35464f2bf270433d60e873fd540edd